File: articles/calculus-for-systems-modeling/articles/economic-growth-and-adjustment-models/c/economic_growth_adjustment.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

double exponential_output(double y0, double g, double t){
  return y0 * exp(g * t);
}

int main(void){
  double rates[] = {0.01, 0.025, 0.04};
  printf("scenario_name,model_type,growth_rate,final_output,doubling_time,warning\n");
  for(int i=0;i<3;i++){
    double g = rates[i];
    printf("growth_rate_case,exponential_growth,%.6f,%.6f,%.6f,growth_rate_assumptions_compound\n", g, exponential_output(100.0, g, 40.0), log(2.0)/g);
  }
  return EXIT_SUCCESS;
}

File: articles/calculus-for-systems-modeling/articles/economic-growth-and-adjustment-models/fortran/economic_growth_adjustment.f90
program economic_growth_adjustment
  implicit none
  real(8), dimension(3) :: rates
  real(8) :: g
  integer :: i
  rates = (/0.01d0, 0.025d0, 0.04d0/)
  print '(A)', 'scenario growth_rate final_output doubling_time'
  do i = 1, 3
    g = rates(i)
    print '(I0,1X,F12.6,1X,F12.6,1X,F12.6)', i, g, 100.0d0 * exp(g * 40.0d0), log(2.0d0)/g
  end do
end program
